### PR TITLE
SPLASH-29: Auth stack UI

### PR DESCRIPTION
**Pull request by Pepegaga**
Login, register and logout with mocks added

### CODE REVIEW

File: src/app/authentication/reducers.ts [hunk 1]
```diff
@@ -18,15 +18,15 @@ const unpersistedReducer = createReducer<AuthenticationState>(initialState)
         error: undefined,
         isFetching: true,
         token: undefined,
-        username: action.payload.username,
+        username: action.payload.email,
```
Comment (TheLonelyAstronaut) [inline src/app/authentication/reducers.ts]: Change to email

File: src/app/authentication/routing.tsx [hunk 2]
```diff
@@ -12,7 +12,7 @@ const Stack = createStackNavigator<AuthParamList>();
 
 export const AuthStack: React.FC = () => {
     return (
-        <Stack.Navigator screenOptions={{ headerShown: false }} initialRouteName="Player">
+        <Stack.Navigator screenOptions={{ headerShown: false }} initialRouteName="Login">
             <Stack.Screen name={'Login'} component={LoginScreen} />
             <Stack.Screen name={'Register'} component={RegisterScreen} />
             <Stack.Screen name={'Verification'} component={VerificationScreen} />
```
Comment (TheLonelyAstronaut) [inline src/app/authentication/routing.tsx:18]: Remove verification, not in our plans now

File: src/app/authentication/routing.params.tsx [hunk 3]
```diff
@@ -6,6 +6,7 @@ export type AuthParamList = {
     Register: undefined;
     Verification: undefined;
     Player: undefined;
+    AppTabs: undefined;
```
Comment (TheLonelyAstronaut) [inline src/app/authentication/routing.params.tsx]: Why AppTabs in AuthStack?

File: src/app/authentication/sagas.ts [hunk 4]
```diff
@@ -4,7 +4,11 @@ import { call, put, takeLatest } from 'redux-saga/effects';
 import { client } from '../../graphql/api';
 import { Logger } from '../utils/logger';
 
-import { LOGIN, REGISTER } from './actions';
+import { LOGIN, LOGOUT, REGISTER } from './actions';
+
+export function* logoutSaga(): SagaIterator {
+    yield call(client.logout);
```
Comment (TheLonelyAstronaut) [inline src/app/authentication/sagas.ts:10]: Also you should remove token from redux

File: src/app/authentication/screens/login-screen.component.tsx [hunk 5]
```diff
@@ -15,63 +16,55 @@ export type LoginScreenProps = AuthNavigationProps<'Login'>;
 export const Title = styled.Text`
     color: ${(props) => props.theme.colors.secondary};
     font-size: ${(props) => props.theme.fontSize.large}px;
-    margin-top: 5%;
-    font-weight: ${(props) => props.theme.fontWeight.bold};
     text-align: center;
+    font-family: ${(props) => props.theme.logoFont};
+    margin-top: 12px;
```
Comment (TheLonelyAstronaut) [inline src/app/authentication/screens/login-screen.component.tsx]: Use spacer prop here

File: src/app/authentication/screens/login-screen.component.tsx [hunk 6]
```diff
@@ -15,63 +16,55 @@ export type LoginScreenProps = AuthNavigationProps<'Login'>;
...
-export const Logo = styled.Image`
-    width: 180px;
-    height: 180px;
-    align-self: center;
-    margin-top: ${(props) => props.theme.spacer * 3}%;
+export const LogoContainer = styled.View`
+    width: ${DEVICE_SIZE.width * 0.3}px;
+    height: 40px;
+    background-color: ${(props) => props.theme.colors.main};
+    margin-left: 34px;
```
Comment (TheLonelyAstronaut) [inline src/app/authentication/screens/login-screen.component.tsx]: Spacer prop here

File: src/app/authentication/screens/login-screen.component.tsx [hunk 7]
```diff
@@ -15,63 +16,55 @@ export type LoginScreenProps = AuthNavigationProps<'Login'>;
...
 `;
 
-export const Logo = styled.Image`
-    width: 180px;
-    height: 180px;
-    align-self: center;
-    margin-top: ${(props) => props.theme.spacer * 3}%;
+export const LogoContainer = styled.View`
+    width: ${DEVICE_SIZE.width * 0.3}px;
+    height: 40px;
```
Comment (TheLonelyAstronaut) [inline src/app/authentication/screens/login-screen.component.tsx]: Should be in the theme

File: src/app/authentication/screens/login-screen.component.tsx [hunk 8]
```diff
@@ -84,51 +77,74 @@ export const LoginText = styled.Text`
...
-            </InputArea>
-        </Container>
+        <AvoidingContainer>
+            <KeyboardAvoidingView
+                contentContainerStyle={{ flex: 1 }}
+                style={{ flex: 1 }}
+                behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
+            >
+                {themeKey === ThemesEnum.DARK || ThemesEnum.JAPANESE ? (
+                    <BackgroundImage source={require('../../../assets/background.jpg')} />
```
Comment (TheLonelyAstronaut) [inline src/app/authentication/screens/login-screen.component.tsx]: We have ImageBackground component in React Native. Refactor

File: src/app/authentication/screens/login-screen.component.tsx [hunk 9]
```diff
@@ -84,51 +77,74 @@ export const LoginText = styled.Text`
...
+                <LogoContainer>
+                    <Title>Splash</Title>
+                </LogoContainer>
+                <ScrollView showsVerticalScrollIndicator={false}>
+                    <InputArea>
+                        <EmailText>{I18n.t('Email')}</EmailText>
+                        <Input
+                            //eslint-disable-next-line react-native/no-color-literals
+                            style={{ borderColor: validation ? theme.colors.additivePink : 'red' }}
+                            onChangeText={(val) => {
```
Comment (TheLonelyAstronaut) [inline src/app/authentication/screens/login-screen.component.tsx]: Move to useCalback

File: src/app/authentication/screens/login-screen.component.tsx [hunk 10]
```diff
@@ -84,51 +77,74 @@ export const LoginText = styled.Text`
...
+                                setEmail(val);
+                                setValidation(validateEmail(email));
+                            }}
+                        />
+                        <InputText>{I18n.t('Password')}</InputText>
+                        <Input secureTextEntry={true} onChangeText={(val) => setPassword(val)} />
```
Comment (TheLonelyAstronaut) [inline src/app/authentication/screens/login-screen.component.tsx]: useCallback

File: src/app/authentication/screens/login-screen.component.tsx [hunk 11]
```diff
@@ -84,51 +77,74 @@ export const LoginText = styled.Text`
...
+                        <LinearButton title={'SignIn'} onPress={handleLogin} />
+                        <SignUpWrapper onPress={() => props.navigation.navigate('Register')}>
```
Comment (TheLonelyAstronaut) [inline src/app/authentication/screens/login-screen.component.tsx:154]: useCallback

File: src/app/authentication/screens/register-screen.component.tsx [hunk 12]
```diff
@@ -1,63 +1,106 @@
...
+                        <Title>Splash</Title>
+                    </LogoContainer>
+                </LogoWrapper>
+                <ScrollView showsVerticalScrollIndicator={false}>
+                    <RegisterInputArea>
+                        <EmailText>{I18n.t('Name')}</EmailText>
+                        <Input onChangeText={(val) => setName(val)} />
+                        <InputText>{I18n.t('Email')}</InputText>
+                        <Input
+                            //eslint-disable-next-line react-native/no-color-literals
```
Comment (TheLonelyAstronaut) [inline src/app/authentication/screens/register-screen.component.tsx]: Move color to theme :p

File: src/app/home/screens/home-screen.component.tsx [hunk 13]
```diff
@@ -28,12 +31,16 @@ export const HomeScreenComponent: React.FC = () => {
...
     const handleRegister = () => {
-        dispatch(REGISTER.TRIGGER({ username: 'vlad', login: 'sfvb', password: '234342' }));
+        dispatch(REGISTER.TRIGGER({ username: 'vlad', email: 'sfvb', password: '234342' }));
     };
 
     const handleSearch = () => {
         dispatch(SEARCH_ALL.TRIGGER('Sunflower'));
     };
+    const handleLogout = () => {
+        dispatch(LOGOUT.TRIGGER());
```
Comment (TheLonelyAstronaut) [inline src/app/home/screens/home-screen.component.tsx:33]: WTF is this?)))

File: src/graphql/api.ts [hunk 14]
```diff
@@ -44,25 +44,31 @@ export class GraphQLAPI {
     };
 
     login = async (payload: LoginPayload): Promise<AuthCompletedPayload> => {
-        const user = users.filter((user) => user.username === payload.username && user.password === payload.password);
+        const user = users.filter((user) => user.email === payload.email && user.password === payload.password);
+
         if (user.length > 0) {
             this.setAuthToken(user[0].token);
-            return { username: user[0].username, token: user[0].token };
+            console.log(user);
```
Comment (TheLonelyAstronaut) [inline src/graphql/api.ts]: Remove logging))0

File: src/locales/default-locale.ts [hunk 15]
```diff
@@ -1,4 +1,8 @@
 export interface DefaultLocale {
-    login: string;
-    hello: string;
+    Email: string;
```
Comment (TheLonelyAstronaut) [inline src/locales/default-locale.ts]: Do something like

auth: {
email: string;
}
home: {
....
}

File: src/app/home/screens/home-screen.component.tsx (hunk 13 above)
Comment (TheLonelyAstronaut) [reply src/app/home/screens/home-screen.component.tsx:33]: I mean next line))0

File: src/app/authentication/routing.tsx (hunk 2 above)
Comment (TheLonelyAstronaut) [reply src/app/authentication/routing.tsx:18]: + {'String literal'} <- refactor constant in line 15

File: src/app/authentication/screens/register-screen.component.tsx (hunk 12 above)
Comment (TheLonelyAstronaut) [reply src/app/authentication/screens/register-screen.component.tsx]: BTW you don't need to provide style prop for style component. You should add custom props when you creating it:

```
const Bruh = styled.View<{isValid: boolean}>`
    background-color: ${isValid ? ... : ...}
`;

...

<Bruh isValid={isValidState}/>
```

File: src/app/authentication/screens/login-screen.component.tsx (hunk 8 above)
Comment (Pepegaga) [reply src/app/authentication/screens/login-screen.component.tsx]: > 

this is it

File: src/app/authentication/screens/login-screen.component.tsx (hunk 9 above)
Comment (Pepegaga) [reply src/app/authentication/screens/login-screen.component.tsx:133]: if I move it useCallback it not working

File: src/app/authentication/authentication.types.ts [hunk 16]
```diff
@@ -3,26 +3,29 @@ export interface AuthenticationState {
...
     username?: string;
     token?: string;
+    email?: string;
 }
 
 export interface LoginPayload {
-    username: string;
+    email: string;
     password: string;
+    username?: string;
```
Comment (TheLonelyAstronaut) [inline src/app/authentication/authentication.types.ts]: Why we need username in login, if we using email?

File: src/app/authentication/authentication.types.ts [hunk 17]
```diff
@@ -3,26 +3,29 @@ export interface AuthenticationState {
...
 export interface RegisterPayload {
     username: string;
     password: string;
-    login: string;
+    email: string;
 }
 
 export interface AuthCompletedPayload {
+    email: string;
+    token?: string;
```
Comment (TheLonelyAstronaut) [inline src/app/authentication/authentication.types.ts]: Why token with ? mark? We must send auth token on server side

File: src/app/authentication/authentication.types.ts [hunk 18]
```diff
@@ -3,26 +3,29 @@ export interface AuthenticationState {
...
     username: string;
-    token: string;
 }
 
 export interface RegisterCompletedPayload {
-    username: string;
+    email: string;
     login: string;
     password: string;
```
Comment (TheLonelyAstronaut) [inline src/app/authentication/authentication.types.ts:27]: We are storing password on the phone? Bad idea.

File: src/app/authentication/screens/login-screen.component.tsx (hunk 8 above)
Comment (TheLonelyAstronaut) [reply src/app/authentication/screens/login-screen.component.tsx]: You can add ternar operator on source prop

File: src/app/authentication/screens/login-screen.component.tsx (hunk 11 above)
Comment (TheLonelyAstronaut) [reply src/app/authentication/screens/login-screen.component.tsx:154]: useCallback

File: src/app/authentication/screens/register-screen.component.tsx [hunk 19]
```diff
@@ -1,63 +1,127 @@
...
+                style={{ flex: 1 }}
+                behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
+            >
+                {themeKey === ThemesEnum.DARK && ThemesEnum.JAPANESE ? (
+                    <BackgroundImage source={require('../../../assets/background.jpg')} />
+                ) : (
+                    <BackgroundImage source={require('../../../assets/light-background.jpg')} />
+                )}
+                <LogoWrapper>
+                    <BackButtonContainer onPress={() => props.navigation.goBack()}>
```
Comment (TheLonelyAstronaut) [inline src/app/authentication/screens/register-screen.component.tsx]: useCallback

File: src/app/authentication/screens/register-screen.component.tsx [hunk 20]
```diff
@@ -1,63 +1,127 @@
...
-            </RegisterInputArea>
-        </Container>
+        <AvoidingContainer>
+            <KeyboardAvoidingView
+                contentContainerStyle={{ flex: 1 }}
+                style={{ flex: 1 }}
+                behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
+            >
+                {themeKey === ThemesEnum.DARK && ThemesEnum.JAPANESE ? (
+                    <BackgroundImage source={require('../../../assets/background.jpg')} />
```
Comment (TheLonelyAstronaut) [inline src/app/authentication/screens/register-screen.component.tsx]: ternar operator in source prop

File: src/app/authentication/screens/login-screen.component.tsx [hunk 21]
```diff
@@ -84,51 +85,77 @@ export const LoginText = styled.Text`
...
-        );
-    }, [dispatch]);
+    const [email, setEmail] = useState('');
+    const [password, setPassword] = useState('');
+    const [validation, setValidation] = useState(true);
 
-    const handleTheme = useCallback(() => {
-        dispatch(CHANGE_THEME({ theme: ThemesEnum.LIGHT }));
-    }, [dispatch]);
+    const handleLogin = () => {
```
Comment (TheLonelyAstronaut) [inline src/app/authentication/screens/login-screen.component.tsx]: useCallback))0

File: src/app/home/screens/home-screen.component.tsx [hunk 22]
```diff
@@ -28,12 +31,15 @@ export const HomeScreenComponent: React.FC = () => {
     }, [dispatch]);
 
     const handleRegister = () => {
```
Comment (TheLonelyAstronaut) [inline src/app/home/screens/home-screen.component.tsx:30]: Remove this from homescreen

File: src/app/home/screens/home-screen.component.tsx [hunk 23]
```diff
@@ -28,12 +31,15 @@ export const HomeScreenComponent: React.FC = () => {
...
-        dispatch(REGISTER.TRIGGER({ username: 'vlad', login: 'sfvb', password: '234342' }));
+        dispatch(REGISTER.TRIGGER({ username: 'vlad', email: 'sfvb', password: '234342' }));
     };
 
     const handleSearch = () => {
```
Comment (TheLonelyAstronaut) [inline src/app/home/screens/home-screen.component.tsx:34]: And this

File: src/app/authentication/sagas.ts [hunk 24]
```diff
@@ -4,7 +4,12 @@ import { call, put, takeLatest } from 'redux-saga/effects';
...
+    yield put(LOGOUT.COMPLETED());
+}
 
 export function* registerSaga(action: ReturnType<typeof REGISTER.TRIGGER>): SagaIterator {
     yield call(client.register, action.payload);
```
Comment (TheLonelyAstronaut) [inline src/app/authentication/sagas.ts:10]: After register we should automatically login, so we should return token, email and username from server and set it to redux

File: src/app/authentication/authentication.types.ts [hunk 25]
```diff
@@ -3,26 +3,22 @@ export interface AuthenticationState {
...
 
 export interface RegisterPayload {
     username: string;
     password: string;
-    login: string;
+    email: string;
 }
 
 export interface AuthCompletedPayload {
-    username: string;
```
Comment (TheLonelyAstronaut) [inline src/app/authentication/authentication.types.ts:20]: Create type User, that will contain all this field. Add username field to User type. 
Username can be //Владислав//, you will display this data in profile. And AuthPayload will be
{
data: User;
}

File: src/app/authentication/sagas.ts [hunk 26]
```diff
@@ -4,10 +4,17 @@ import { call, put, takeLatest } from 'redux-saga/effects';
...
```
Comment (TheLonelyAstronaut) [inline src/app/authentication/sagas.ts:10]: Why you calling register twice? Also why you don't have .failed handling, example in loginSaga

**Review by TheLonelyAstronaut — APPROVED** [review]

**Pull request merged**